odoo/addons/base/models/ir_asset.py

```diff
@@ -305,9 +305,6 @@ def _get_paths(self, path_def, installed, extensions=None):
 
         safe_path = True
         if addon_manifest:
-            if addon not in installed:
-                # Assert that the path is in the installed addons
-                raise Exception("Unallowed to fetch files from addon %s" % addon)
```

Comment:
Just this patch is enough to prevent Internal Server error.

However, the impatient user will still face few issues:

1. Configurator will ask to select the theme again. 
2. If user select the theme again, they may get error "Modules operations are not possible this time..."
3. Anyway, the user will never be redirected to next step (Website Editor)

![image](https://user-images.githubusercontent.com/186131/230594175-9c6cb89d-ded1-45ac-bf75-8bb1256b3490.png)

cc @odoo/rd-security

Reply:
Preventing errors for users refreshing the page does not seem a valid reason to remove this security check

Reply:
The error may happen if another user opens backend while theme is installed, but the other modules installation is not finished